tests/framework/extensions/rke1/nodepools/nodepools.go

```diff
@@ -54,3 +60,52 @@ func RKE1NodePoolSetup(client *rancher.Client, nodeRoles []NodeRoles, ClusterID,
 
 	return &nodePoolConfig, nil
 }
+
+// ScaleWorkerNodePool is a helper method that will add a worker node pool to the existing RKE1 cluster. Once done, it will scale
+// the worker node pool to add a worker node, scale it back down to remove the worker node, and then delete the worker node pool.
+func ScaleWorkerNodePool(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, NodeTemplateID string) error {
+	nodePoolConfig := management.NodePool{
+		ClusterID:               ClusterID,
+		ControlPlane:            false,
+		DeleteNotReadyAfterSecs: 0,
+		Etcd:                    false,
+		HostnamePrefix:          "auto-rke1-scale-test",
+		NodeTemplateID:          NodeTemplateID,
+		Quantity:                1,
+		Worker:                  true,
+	}
+
+	nodePool, err := client.Management.NodePool.Create(&nodePoolConfig)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(time.Second * time.Duration(longTimeout))
+
+	nodePoolConfig.Quantity = 2
+
+	updatedPool, err := client.Management.NodePool.Update(nodePool, &nodePoolConfig)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(time.Second * time.Duration(longTimeout))
```

Comment:
As I understand, this is here because we don't have a good `wait.watch` for rke1 at the moment. Could we at least validate that the scaled pool has all nodes active?

Reply:
Correct to your inquiry. The one solution I thought would work best that I attempted was the following:

- Create node pool struct with only worker node defined
- Create worker node pool
- Create node struct that references the worker node pool that was just created (I was doing this via the struct's `nodePoolID` parameter
- Create a worker node
- Track the status via the `State` parameter found in the worker node struct

The issue with the above is that the worker node does not find its way into the node pool that it is associated with....but that is IMO the simplest way to track if nodes are in a state of `Active`.

Reply:
As an aside, we have `IsHostedClusterProvisioningReady` in `clusters.go` that I tried to use for both `v3.Node` and `v3.NodePool` objects, but that was not working either.

Reply:
Markus you would need to create your own check func. Do something like this 
`watchInterface, err := r.client.GetManagementWatchInterface(management.NodePoolType, opts)
  require.NoError(r.T(), err)
  `
Then create your own nodePoolReady function. Use `IsHostedClusterProvisioningReady` as reference 

I'm not sure exactly how it would look like because I haven't looked but I assume something like this 

```
func IsNodePoolrReady(event watch.Event) (ready bool, err error) {
	nodePoolUnstructured := event.Object.(*unstructured.Unstructured)
	nodePool := &v3.NodePool{}
	err = scheme.Scheme.Convert(nodePoolUnstructured, nodePool, nodePoolUnstructured())
	if err != nil {
		return false, err
	}
	// check some status or something here. 
	}

	return false, nil
}
```

Reply:
Oh ok coo! Sounds good. What you can also do is this 

```
return kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
		resp, err := client.Management.NodePool("id")
	})
```

And get the node pool until that status is active or whatever the condition is like how you described.
import this package `kwait "k8s.io/apimachinery/pkg/util/wait"`

Reply:
Implemented fix here that removes the time.Sleeps when checking node status. I did keep one as I'm scaling down. This is to allow Rancher some "breathing" room. Without it, it will immediately move to delete the node pool and continue with the next test.